pkg/scheduler/api/resource_info.go

```diff
@@ -224,39 +224,35 @@ func (r *Resource) Multi(ratio float64) *Resource {
 
 // Less checks whether a resource is less than other
 func (r *Resource) Less(rr *Resource) bool {
-	lessFunc := func(l, r float64) bool {
-		return l < r
+	greaterFunc := func(l, r float64) bool {
+		return l > r
 	}
 
-	if !lessFunc(r.MilliCPU, rr.MilliCPU) {
+	if greaterFunc(r.MilliCPU, rr.MilliCPU) {
 		return false
 	}
-	if !lessFunc(r.Memory, rr.Memory) {
+
+	if greaterFunc(r.Memory, rr.Memory) {
 		return false
 	}
 
-	if r.ScalarResources == nil {
-		if rr.ScalarResources != nil {
-			for _, rrQuant := range rr.ScalarResources {
-				if rrQuant <= minMilliScalarResources {
-					return false
-				}
+	same := true
+	same = same && (r.MilliCPU == rr.MilliCPU)
+	same = same && (r.Memory == rr.Memory)
+	if r.ScalarResources != nil && rr.ScalarResources != nil {
+		for rName, rQuant := range r.ScalarResources {
+			rrQuant := rr.ScalarResources[rName]
+			if greaterFunc(rQuant, rrQuant) {
+				return false
 			}
+
+			same = same && (rQuant == rrQuant)
 		}
-		return true
 	}
-
-	if rr.ScalarResources == nil {
+	if same {
```

Comment:
How about the following testcase?
`
r = {
cpu: 100,
memory: 0,
scalar_resource:{
   gpu:1
}
}
rr = {
cpu: 100,
memory: 100
}
`

Reply:
result will be `false`
Why should care about such small resource?

Reply:
I'm sorry, it's a bug, I'll fix it

Reply:
> I'm sorry, it's a bug, I'll fix it

In my opinion, current logic has serious problem for not considering some boundary testcases. Fix on current code is not a good idea.